distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -281,3 +102,149 @@ type Marshalling_Error
             "Type mismatch error: the json with type `" + json_text + "` did not match the format `" + format_text + "`."
         Marshalling_Error.Missing_Field _ field _ ->
             "Missing field in Json: the field `" + field.to_text "` was missing in the json."
+
+type JS_Object
+    ## Creates a JS_Object from a list of key-value pairs.
+       Keys must be `Text` values.
+       Values will be recursively converted to JSON serializable as needed.
+    from_pairs : Vector -> JS_Object
+    from_pairs pairs =
+        js_object = pairs.fold new_object current->pair->
+            case pair.at 0 of
+                text : Text ->
+                    set_value current text (pair.at 1).to_js_object
+                    current
+                _ -> Error.throw (Illegal_Argument.Error "JS_Object.from_pairs: key must be a Text value")
+        JS_Object.Value js_object
+
+    ## PRIVATE
+    Value js_object
+
+    ## Get a value for a key of the object, or a default value if that key is not present.
+
+       Arguments:
+       - key: The key to get.
+       - if_missing: The value to return if the key is not found.
+    get : Text -> Any -> JS_Object | Boolean | Number | Nothing | Text | Vector
+    get self key ~if_missing=Nothing =
+        if (has_property self.js_object key) . not then if_missing else
+            value = get_value self.js_object key
+            make_enso value
+
+    ## Get a value for a key of the object.
+       If the key is not found, throws a `No_Such_Key` error.
+
+       Arguments:
+       - key: The key to get.
+    at : Text -> JS_Object | Boolean | Number | Nothing | Text | Vector
+    at self key = self.get key (Error.throw (No_Such_Key.Error self key))
+
+    ## Get the keys of the object
+    field_names : Vector
+    field_names self =
+        Vector.from_polyglot_array (get_property_names self.js_object)
+
+    ## Gets the number of keys in the object
+    length : Number
+    length self =
+        get_property_names self.js_object . length
+
+    ## Convert the object to a Vector of Pairs
+    to_vector : Vector
+    to_vector self =
+        keys = get_property_names self.js_object
+        proxy = Array_Proxy.new keys.length (i-> Pair.new (keys.at i) (self.get (keys.at i)))
+        Vector.from_polyglot_array proxy
+
+    ## Convert the object to a JSON
+    to_js_object : JS_Object
+    to_js_object self = self
```

Comment:
I think you told me that `to_json` converts to a JSON text representation while `to_js_object` converts to a _JS Object_ **not** _JSON_ (which is a textual notations for the former).

Can we amend the doc to be consistent?
```suggestion
    ## Convert the object to a JS Object.
    to_js_object : JS_Object
    to_js_object self = self
```

Also please ensure we finish doc strings with dots.

Should `JS_Object` also have a `to_json = self.to_text`?